docs/versioning.md

```diff
@@ -0,0 +1,566 @@
+# Versioning
+
+This section describes S3 Server's support for the specific AWS S3 Bucket
+Versioning feature.
+
+## AWS S3 Bucket Versioning
+
+An object in a versioning enabled bucket can have multiple versions, whereas an
+object can have only a single version in a standard bucket without versioning.
+
+With the standard S3 API, any update (PUT, DELETE) to an object in a versioning
```

Comment:
It could be worth mentioning in this paragraph that a DELETE on a versioned bucket creates delete markers in place of deleting the data, to precise the "create a new version" for the DELETE case, as it seems the first mention of delete markers is far below in this document.